src/components/TodoApp/TodoApp.jsx

```diff
@@ -0,0 +1,163 @@
+import React, { useState, useMemo, useCallback } from 'react';
+import { TodoList } from '../TodoList/TodoList';
+import { TodoFilter } from '../TodoFilter/TodoFilter';
+import { useLocalStorage } from '../../styles/localStorage';
+
+export const TodoApp = () => {
+  const [title, setTitle] = useState('');
+  const [activeFilter, setActiveFilter] = useState('All');
+  const [todos, setTodos] = useLocalStorage('todos', []);
+
+  const switchFilter = useCallback((filter) => {
+    let visibleTodos;
+
+    switch (filter) {
+      case 'Active':
+        visibleTodos = todos.filter(todo => !todo.completed);
+        break;
+
+      case 'Completed':
+        visibleTodos = todos.filter(todo => todo.completed);
+        break;
+
+      case 'All':
+        visibleTodos = todos;
+        break;
+
+      default:
+        visibleTodos = todos;
+    }
+
+    return visibleTodos;
+  }, [todos]);
+
+  const filteredTodos = useMemo(
+    () => switchFilter(activeFilter),
+    [switchFilter, activeFilter],
+  );
+
+  const filterChange = (filter) => {
+    setActiveFilter(filter);
+    switchFilter(filter);
+  };
+
+  const toggleAllTodos = () => {
+    let toggledTodos;
+
+    if (todos.some(todo => todo.completed === false)) {
+      toggledTodos
+        = todos.map(todo => ({
+          ...todo,
+          completed: true,
+        }));
+    } else {
+      toggledTodos
+        = todos.map(todo => ({
+          ...todo,
+          completed: false,
+        }));
+    }
+
+    setTodos(toggledTodos);
+  };
+
+  const handleInputChange = (event) => {
+    const { value } = event.target;
+
+    setTitle(value);
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+
+    addTodo(title);
+    setTitle('');
+  };
+
+  const addTodo = (newtitle) => {
+    if (newtitle !== '') {
+      const newTodo = {
+        id: +new Date(),
+        title: newtitle,
+        completed: false,
+      };
+
+      setTodos([...todos, newTodo]);
+    }
+  };
+
+  const deleteTodo = (todoId) => {
+    setTodos(prevTodos => prevTodos.filter(
+      todo => todo.id !== todoId,
+    ));
+  };
+
+  const toggleTodoComplete = (todo) => {
+    const selectedTodo = todos.find(selected => todo.id === selected.id);
+
+    selectedTodo.completed = !selectedTodo.completed;
+    setTodos([...todos]);
+  };
```

Comment:
это нужно делать так, и это есть у Миши в видео
```suggestion
  const toggleTodoComplete = (todoId) => {
    setTodos((prevTodo) => prevTodo.map(todo => {
      if (todo.id !== todoId) {
        return todo;
      }
      
      return {
        ...todo,
        completed: !todo.completed,
      };
    }));
  };
```